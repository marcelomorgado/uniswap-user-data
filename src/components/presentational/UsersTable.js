import React from "react";
import PropTypes from "prop-types";
import VirtualizedTable from "./VirtualizedTable";
import Paper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";

function UsersTable({ rows, onRowClick, classes }) {
  const columns = [
    {
      width: 200,
      flexGrow: 1.0,
      label: "User ID",
      dataKey: "id",
    },
    {
      width: 120,
      label: "ETH Balance",
      dataKey: "etherBalance",
      numeric: false,
    },
  ];

  return (
    <Paper className={classes.paper}>
      <VirtualizedTable
        rowCount={rows.length}
        rowGetter={({ index }) => rows[index]}
        onRowClick={e => {
          onRowClick(e);
        }}
        columns={columns}
      />
    </Paper>
  );
}

UsersTable.propTypes = {
  onRowClick: PropTypes.func.isRequired,
  rows: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      etherBalance: PropTypes.string.isRequired,
    })
  ).isRequired,
  classes: PropTypes.object.isRequired,
};

const styles = () => ({
  paper: { height: 400, width: "100%" },
});

export default withStyles(styles)(UsersTable);
