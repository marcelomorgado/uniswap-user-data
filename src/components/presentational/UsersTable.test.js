import React from "react";
import { shallow } from "enzyme";
import UsersTable from "./UsersTable";

it("renders without crashing", () => {
  const rows = [
    {
      id: "0x0000000000c90bc353314b6911180ed7e06019a9",
      etherBalance: "1.234567890",
    },
    {
      id: "0x000000000a3c7c86345a35a0e97a4bb4370a8dd9",
      etherBalance: "1000.98787878787",
    },
  ];
  expect(shallow(<UsersTable rows={rows} />)).toMatchSnapshot();
});
