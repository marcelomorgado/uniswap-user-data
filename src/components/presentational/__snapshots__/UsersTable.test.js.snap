// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<UsersTable
  classes={
    Object {
      "paper": "UsersTable-paper-1",
    }
  }
  rows={
    Array [
      Object {
        "etherBalance": "1.234567890",
        "id": 0,
        "userId": "0x0000000000c90bc353314b6911180ed7e06019a9",
      },
      Object {
        "etherBalance": "1000.98787878787",
        "id": 1,
        "userId": "0x000000000a3c7c86345a35a0e97a4bb4370a8dd9",
      },
    ]
  }
/>
`;
